### PR TITLE
[PVM] MultisigAliasTx implementation

### PR DESCRIPTION
**Pull request by charalarg**
## Description ##
This PR adds the MultisigAliasTx implementation on the js library in order to be able to be used from the camino-suite. The equivalent implementation of the MultisigAliasTx on the node is described in this [PR](https://github.com/chain4travel/caminogo/pull/246) 

## Changes ##
- Implementation of the MultisigAliasTx Class that handles the tx logic.
- Implementation of the MultisigAlias Class which represents the alias definition in the tx.
- Added api and builder functions.
- Added unit tests.
- Added examples for both creating a new multisig alias definition (single sig tx) and updating an already existing ones (multisig tx).

### CODE REVIEW

File: tests/apis/platformvm/multisigaliastx.test.ts [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+import { ZeroBN } from "src/common"
+import Avalanche from "src/index"
+
+const avalanche: Avalanche = new Avalanche(
+  "127.0.0.1",
+  9650,
+  "https",
+  12345,
+  undefined,
+  undefined
```
Comment (knikos) [inline tests/apis/platformvm/multisigaliastx.test.ts]: these 2 last parameters can be omitted

File: tests/apis/platformvm/multisigaliastx.test.ts [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    Buffer.from("memo"),
+    new MultisigAlias(
+      undefined,
+      Buffer.from(aliasMemo, "utf-8"),
+      new ParseableOutput(
+        new SECPOwnerOutput(
+          outputOwners.getAddresses(),
+          outputOwners.getLocktime(),
+          outputOwners.getThreshold()
+        )
```
Comment (knikos) [inline tests/apis/platformvm/multisigaliastx.test.ts]: ` new SECPOwnerOutput(
          outputOwners.getAddresses(),
          outputOwners.getLocktime(),
          outputOwners.getThreshold()
        )`
could be replaced by
`new ParseableOutput(outputOwners),`

File: tests/apis/platformvm/multisigaliastx.test.ts [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    multisigAliasTx.addSignatureIdx(0, keychain.getAddresses()[0])
+    const lastSigIdx =
+      multisigAliasTx.getSigIdxs()[multisigAliasTx.getSigIdxs().length - 1]
+    expect(lastSigIdx["source"]).toStrictEqual(keychain.getAddresses()[0])
+  })
+
+  test("getAuth", async (): Promise<void> => {
+    const actualAuth: SubnetAuth = multisigAliasTx.getAuth()
+    expect(actualAuth).toStrictEqual(auth)
+  })
```
Comment (knikos) [inline tests/apis/platformvm/multisigaliastx.test.ts:167]: could you add a unit test maybe for `getMultisigAlias` or do you think it doesn't make sense?

File: src/apis/platformvm/multisigaliastx.ts [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    this.memo = memo
+  }
+
+  getOwners(): ParseableOutput {
+    return this.owners
+  }
+
+  setOwners(owners: ParseableOutput): void {
+    this.owners = owners
+  }
```
Comment (knikos) [inline src/apis/platformvm/multisigaliastx.ts]: unused setters and getters can be completely removed and added if and when needed

File: examples/common/examplesConfig.json [hunk 5]
```diff
@@ -2,5 +2,5 @@
   "host": "localhost",
   "port": 9650,
   "protocol": "http",
-  "networkID": 12345
+  "networkID": 1002
```
Comment (knikos) [inline examples/common/examplesConfig.json]: i would recommend a new json called `examplesCaminoConfig.json` to be created and referenced. and this change to be reverted

File: src/apis/platformvm/api.ts [hunk 6]
```diff
@@ -2697,6 +2698,69 @@ export class PlatformVMAPI extends JRPCAPI {
...
+  /**
+   * Build an unsigned [[MultisigAliasTx]].
+   *
+   * @param utxoset A set of UTXOs that the transaction is built on
+   * @param fromAddresses The addresses being used to send the funds from the UTXOs {@link https://github.com/feross/buffer|Buffer}
+   * @param changeAddresses The addresses that can spend the change remaining from the spent UTXOs.
+   * @param memo Optional contains arbitrary bytes, up to 256 bytes
+   * @param asOf Optional. The timestamp to verify the transaction against as a {@link https://github.com/indutny/bn.js/|BN}
+   * @param changeThreshold Optional. The number of signatures required to spend the funds in the resultant change UTXO
+   * @param multisigAliasParams An object containing the parameters for the multisigAliasTx
```
Comment (knikos) [inline src/apis/platformvm/api.ts]: maybe reorder parameters to match the function's signature

**Review by knikos — CHANGES_REQUESTED** [review]

**Review by knikos — APPROVED** [review]

File: examples/common/examplesConfig.json [hunk 7]
```diff
@@ -2,5 +2,5 @@
   "host": "localhost",
   "port": 9650,
   "protocol": "http",
-  "networkID": 12345
```
Comment (c4t-ag) [inline examples/common/examplesConfig.json]: Pls. revert

File: examples/common/examplesKopernikusConfig.json [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (c4t-ag) [inline examples/common/examplesKopernikusConfig.json]: Pls. revert

**Review by c4t-ag — CHANGES_REQUESTED** [review]

**Review by c4t-ag — APPROVED** [review]

**Pull request merged**